# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/security/trusted-types/TrustedTypePolicyFactory.cpp
# Commit: 178d62ae39b5
# Full Hash: 178d62ae39b57dcc9c17ccc8130c6f57d2ecc289
# Author: Mirko Brodesser <mbrodesser@igalia.com>
# Date: 2024-06-25 21:32:30
# Description:
#   Bug 1903879: check inner window's nullness in `TrustedTypePolicyFactor::createPolicy`. r=tschuster
#   
#   The inner window may be null for workers.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D214708
# ==============================================================================

diff -r c6d2db406739 -r 178d62ae39b5 dom/security/trusted-types/TrustedTypePolicyFactory.cpp
--- a/dom/security/trusted-types/TrustedTypePolicyFactory.cpp	Tue Jun 25 08:43:14 2024 +0000
+++ b/dom/security/trusted-types/TrustedTypePolicyFactory.cpp	Tue Jun 25 08:43:14 2024 +0000
@@ -26,7 +26,10 @@
   // CSP-support for Workers will be added in
   // <https://bugzilla.mozilla.org/show_bug.cgi?id=1901492>.
   // That is, currently only Windows are supported.
-  nsIContentSecurityPolicy* csp = mGlobalObject->GetAsInnerWindow()->GetCsp();
+  nsIContentSecurityPolicy* csp =
+      mGlobalObject->GetAsInnerWindow()
+          ? mGlobalObject->GetAsInnerWindow()->GetCsp()
+          : nullptr;
 
   if (csp) {
     uint32_t numPolicies = 0;